sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java

```diff
@@ -51,10 +52,17 @@ public ScreenshotEventProcessor(
     application.registerActivityLifecycleCallbacks(this);
   }
 
+  public void setCurrentActivity(@NonNull Activity activity) {
+    if (currentActivity != null && currentActivity.get() == activity) {
+      return;
+    }
+    currentActivity = new WeakReference<>(activity);
+  }
+
   @SuppressWarnings("NullAway")
   @Override
   public @NotNull SentryEvent process(final @NotNull SentryEvent event, @NotNull Hint hint) {
-    if (!lifecycleCallbackInstalled) {
+    if (!lifecycleCallbackInstalled || !HintUtils.shouldApplyScopeData(hint)) {
```

Comment:
H: `EnvelopeFileObserver` uses the `CachedEnvelopeHint` hint so this won't work, we need to extend the current hint type for Hybrid SDKs most likely.
The goal here is that at least for screenshots, screenshots should not be taken (again) for Hybrid SDKs.

Reply:
Okay got it, that was my goal, I've just misunderstood the function.